src/main/java/com/streamr/client/utils/HttpUtils.java

```diff
@@ -18,6 +19,7 @@
     // Thread safe
     public static final Moshi MOSHI = new Moshi.Builder()
             .add(Date.class, new StringOrMillisDateJsonAdapter().nullSafe())
+            .add(BigDecimal.class, new BigDecimalAdapter())
```

Comment:
You should be able to remove all the null checking code in `BigDecimalAdapter` if you do like the above line 21 and call `nullSafe()` to create a null-checking wrapper to the adapter

```suggestion
            .add(BigDecimal.class, new BigDecimalAdapter().nullSafe())
```